### ISSUE TITLE
Introduce Truncate

### ISSUE THREAD

**Issue by Jan-Eimertenbrink** [mittwald/flow#2111]
## Description
We will eventually face the issue of extremely long titles appearing in the `ContentHeader` in mStudio. From an accessibility perspective, text should always remain fully readable. However, long titles may cause the layout to break and extend beyond the intended blue wave area.

To prevent this, I propose introducing a **Truncate** option within our DesignSystem. This would allow titles to visually shorten with an ellipsis when necessary, helping maintain layout stability.

## Accessibility Considerations

Truncation should be **used carefully** and only when it does not prevent users from accessing the full information.  
While truncating text with an ellipsis can exclude keyboard users from reading the full content via hover tooltips, the approach can still be considered accessible if keyboard users have another way to access the complete information.

Example from the current mStudio:
- The title can be truncated in the `ContentHeader`.
- The full title remains accessible through the **LabeledValues** on the detail page.

<img width="1016" height="251" alt="Image" src="https://github.com/user-attachments/assets/b9398618-0618-439c-8cce-566772f05a1d" />

This ensures that all users can still identify the corresponding element while keeping the layout visually consistent.

## Inspiration
We can use [GitHub’s DesignSystem Primer](https://primer.style/product/components/truncate/accessibility/) as a reference, as their style guide provides a comprehensive and well-documented approach to handling text truncation and accessibility.

The introduction of this feature could be postponed until we start refactoring the navigation in mStudio.
